app/frontend/helpers/NamedError.ts

```diff
@@ -1,8 +1,13 @@
-function NamedError(name, message = '', showHelp = '') {
-  const e = new Error(message || name || showHelp)
+type OptionalParams = {
+  message?: string
+  errorStack?: string
+}
+
+function NamedError(name: string, optionalParams: OptionalParams = {}) {
+  const e = new Error(optionalParams.message || name)
   e.name = name
-  e.message = message
-  ;(e as any).showHelp = showHelp
+  e.message = optionalParams.message || ''
+  e.stack = optionalParams.errorStack || e.stack
```

Comment:
Maybe if we wanted Optionalparams to contain (almost) anything like the txPlan, we could implement it like this. Im not that sure that just the stackTrace will help us to solve the bug, so imho adding the txPlan to the error makes sense. Additionally, it could probably be used also in other places.
``` 
function NamedError(name: string, optionalParams: OptionalParams = {}) {
let e = new Error(optionalParams.message || name)
  ;(e as any) = {
    name,
    ...optionalParams,
    stack: optionalParams.errorStack || e.stack,
  }

Reply:
Since optional params happens to be an object, it would be easy to set the whole object as sentry error context you mentioned in #645 .

Reply:
I'm not really sure how safe it is in - I would rather refrain from it if we don't need it for sure